### ISSUE TITLE
Provide migration guide for Artifactory

### ISSUE THREAD

**Issue by codesplode** [dzikoysk/reposilite#1261]
### Request details

First off, thank you for a fantastic project. It felt great replacing a stupidly large machine just for hosting build artificats, with a single core using reposilite. The new setup has not only made builds / dependency retrieval faster, but is embarrassingly lightweight compared to every other solution we've tried. I look forward to contributing more to the project and helping you move Javalin towards coroutines, etc.

For now, I thought I would share some basic information regarding migration of an Artifactory repository, which I just did. I thin a `Migration` section in the guide may help some moving from other repositories, though the process is pretty simple.

Suggested Content for `Migration -> Artifactory`:

## Instructions

1. Export Each Repository from Artifactory.
    Be sure to check the `.m2 compatible export` box to get an export that reposilite can use. Also, we recommend excluding artifactory build metadata if your version of Artifactory has this feature.
2. Move the resulting export files to your new reposilite server and extract them to the repository folder you would like them in.
    Depending on the organization of your new repositories, you may need to move your artifacts out of parent folders to get them in the location you want them in your new repository.
3. Configure the new repository in reposilite, if you haven't already, making sure it is loading files from the correct directory.
    You should be able to browse the artifacts once configuration reloads.

### Combining Releases / Snapshots

In some cases, you may want to merge separated release and snapshot repositories into one Reposilite repository. Rsync offers a useful way to do this and is installed by default on many unix systems. The following command safely merges releases in a `releases` folder into a `snapshots` folder containing snapshots. You can then move / rename the snapshots folder accordingly.

```bash
# Merge all files in releases into the snapshots folder
rsync -avhu --progress ./releases/ ./snapshots/

# Remove the releases folder
rm -rf ./releases

# Move the contents of the snapshots folder to your final target repository
mv ./snapshots/* /data/reposilite/repositories/my-migrated-repository/
```

### Removing Artifactory Build Metadata Folders

If your Artifactory exports contain build metadata folders that you would prefer to remove, you can use the following command to help remove them. This command performs a find for all directories, in the current directory, ending in `.artifactory-metadata`, and removes them from the system.

```bash
rm -rf `find . -type d -name *.artifactory-metadata`
```

**Comment by dzikoysk** [comment]
Nice to hear that! :) Well, I think we won't move to coroutines for now anyway, not only because Javalin does not support this out of the box, but mostly because we want to provide a sane api for plugins written in Java - coroutines pretty much kill such possibility due to the abstraction cost that is hard to handle on Java side. Project Loom might be a better choice in a long term for us.

Speaking of the issue - good idea, I'd also add a `Reposilite 2.x` to the `Migration` section in the future.

**Comment by codesplode** [comment]
Completely understand that. I can appreciate the simplicity of blocking API code as the discussion in Javalin suggest, but we are currently looking to move to Javalin from a Vertx framework we have made that essentially hides vertx under a layer that executes each controller in a coroutine, so there is no direct reference to async or coroutine logic if that makes sense.

Totally agree on project loom. It would still be very easy to stack coroutines or just use Loom's functionality under the hood. As mentioned above, we try to hide non-blocking nature from being a direct concern at the controller layer and instead make use of these features in "services" like our data access layer or email plugins, etc. Really, wrapping the controller execution in a coroutine just provides a seamless layer for the app developer to be able to unknowingly call services that happen to use suspend function in Kotlin as if they were blocking methods.

A whole 'nother conversation for a different thread of course, but I saw and appreciated your comments on the Javalin side very much. A framework is always a mix of tradeoffs and should only ever be about the minimal features to accomplish the goal well. I am a fan of frameworks that move slowly, small amounts of code at a time, without adding a bunch of extra features into the core.
